src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -1225,6 +1248,22 @@ private ClaimsPrincipal ValidateToken(SamlSecurityToken samlToken, string token,
             return new ClaimsPrincipal(identities);
         }
 
+        private static async Task<TokenValidationParameters> PopulateValidationParametersWithCurrentConfigurationAsync(
+            TokenValidationParameters validationParameters)
+        {
+            if (validationParameters.ConfigurationManager != null)
+            {
+                var currentConfiguration = await validationParameters.ConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).ConfigureAwait(false);
+                validationParameters = validationParameters.Clone();
```

Comment:
For ease of reading, i would suggest to create a new variable: TokenValidationParameters validationParametersCloned.

Reply:
Good point. Refactored.